src/scenarios/personalization/ArrayAddFavorites/ArrayAddFavorites.tsx

```diff
@@ -0,0 +1,245 @@
+        query: getTvShowsQuery,
+        context: {
+          headers: {
+            Authorization: `Bearer ${userAccessToken}`,
+          },
+        },
+        fetchPolicy: 'no-cache',
+      });
+
+      const moviesArray = resultMovies.data.movies.nodes;
+      const tvShowsArray = resultTvShows.data.tvshows.nodes;
+
+      // Add Entity Type
+      for (let i = 0; i < moviesArray.length; i++) {
+        moviesArray[i] = Object.assign(moviesArray[i], {
+          type: 'Movie',
+        });
+      }
+
+      for (let i = 0; i < tvShowsArray.length; i++) {
+        tvShowsArray[i] = Object.assign(tvShowsArray[i], {
+          type: 'TV Show',
+        });
+      }
+
+      // Concat Entity Arrays
+      const result = moviesArray.concat(tvShowsArray);
+      setEntityArray(result);
+
+      logger.log(`calling [${fetchAllCatalogItems.name}]`, 'output:', result);
+
+      if (result.errors) {
+        logger.error(result.errors);
+      }
+    } catch (error) {
+      if (error instanceof Error) {
+        logger.error(
+          `calling [${fetchAllCatalogItems.name}]`,
+          'output:',
+          error.message,
+        );
+      }
+    }
+  };
+
+  const addToFavorites = async (): Promise<void> => {
+    try {
+      const apolloClient = getApolloClient(
+        new URL('graphql', activeProfile.personalizationServiceBaseURL).href,
+      );
+
+      const index = entityId.indexOf(':'); // Gets the first index where a colon occurs.
+      const id = entityId.substring(0, index); // Gets the id part
+      const title = entityId.substring(index + 1); // Gets the title part
+
+      interface EntityObject {
+        id: string;
+        title: string;
+      }
+
+      const entityObject: EntityObject = {
+        id: id,
+        title: title,
+      };
+
+      const result = await apolloClient.mutate({
+        mutation: setArrayItemMutation,
+        variables: {
+          input: {
+            scope: 'PROFILE',
+            key: 'user_profile:favorites',
+            value: entityObject,
+          },
+        },
+        context: {
+          headers: {
+            Authorization: `Bearer ${userAccessToken}`,
+          },
+        },
+        fetchPolicy: 'no-cache',
+      });
+
+      if (result.errors) {
+        logger.error(
+          `method [${addToFavorites.name}]`,
+          'output:',
+          result.errors,
+        );
+      } else {
+        logger.log(`method [${addToFavorites.name}]`, 'output:', result.data);
+      }
+    } catch (error) {
+      if (error instanceof Error) {
+        logger.error(
+          `method [${addToFavorites.name}]`,
+          'output:',
+          error.message,
+        );
+      } else {
+        logger.error(
+          `method [${addToFavorites.name}]`,
+          'output:',
+          JSON.stringify(error),
+        );
+      }
+    }
+  };
+
+  return (
+    <Segment basic>
+      <Header size="huge">Array: Add Favorite(s)</Header>
```

Comment:
Update this once you decide which version to use - Favorite(s) or Favorites

Reply:
As discussed will skip this change